tests/generation/test_utils.py

```diff
@@ -1567,13 +1572,13 @@ def test_generate_from_inputs_embeds_decoder_only(self):
             # This test is for decoder-only models (encoder-decoder models have native input embeddings support in the
             # decoder)
             if config.is_encoder_decoder:
-                continue
+                self.skipTest("This test is for decoder-only models")
```

Comment:
We need a continue here :P

With a skip, we skip for all other models in `self.all_generative_model_classes`, and we might have encoder-decoder and decoder-only models mixed up together in that iterator

Reply:
hmm oke, didn't think we could have cases with mixed model types. Was just confused the test was passing for one model while not other, and took me a while to figure out hehe